pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -307,6 +307,12 @@ func (cb *ClusterBuilder) buildCluster(name string, discoveryType cluster.Cluste
 		} else if networkutil.AllIPv6(cb.proxyIPAddresses) {
 			// IPv6 only
 			c.DnsLookupFamily = cluster.Cluster_V6_ONLY
+			// If we are in this mode, Istio sees ourselves as only have IPv6 addresses, but there is actually a link-local
+			// interface that serves the IPv4. Allow both families.
+			// This ensures we do not break DNS resolution to destinations that are IPv4 only.
+			if features.EnableAdditionalIpv4OutboundListenerForIpv6Only {
```

Comment:
This is maybe a dumb question but why wouldn't we _always_ set `c.DnsLookupFamily = cluster.Cluster_ALL` and skip the feature flag?

Reply:
The problem is, consider an IPv4 only pod with no v6 connectivity (or vis-versa). Lets say I am v6 only and calling google.com, which has only both address types. We will load balance between v6 (good) and v4 (bad).

(I think .. this is based on years old memory, worth a shot)

Reply:
Yeah - in my mind this keeps coming back to "you said you wanted V6-only, but you actually want dualstack behavior". 

That's sorta why I feel like the ideal UX for this stuff would just be be `istio.ipv4.enabled: true|false` and `istio.ipv6.enabled: true|false`

For AWS you would set both to true, and this would ~work. For 90% of envs you would set both to true. But if you really know what you're doing, you can turn off one or the other and "go strict" which means we would just fail traffic if you had pods doing v4 stuff and `istio.ipv4.enabled: false`.